trading/trader/order.py

```diff
@@ -66,13 +66,10 @@ def __init__(self, trader):
         self.linked_orders = []
         self.lock = Lock()
 
-    # Disposable design pattern
-    def __enter__(self):
-        self.lock.acquire()
-        return self
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        self.lock.release()
+    # syntax: "async with xxx.get_lock():"
+    # TODO find better way to handle async lock: reuse disposable design pattern ?
```

Comment:
Do you have more informations about this ?

Reply:
I didnt come with a better solution yet, current asyncio.locks are working and are adapter to the new async arch.
However a better solution would be to find a way to use the disposable  design (to be able to use the following syntax like before:)
with order as odr:
    odr.xxx()

with the current solution, we can do:
async with order:
    order.xxx()

but not 
async with order as odr:
    odr.xxx()

Reply:
We should create an issue if we think it's necessary to remove this **with order.get_lock():** syntax, to me it's ok like this and doesn't really have to be fixed but if you think it does then yes.

Reply:
I don't think it changes anythin in performances, I just didn't manage to re-create the previous syntax with **async with** therefore I ended up with this